CallCenterSample/Helpers/TextAnalyticsHelper.cs

```diff
@@ -109,48 +123,41 @@ public static async Task<DetectLanguageResult> GetDetectedLanguageAsync(string i
 
             if (!string.IsNullOrEmpty(input))
             {
-                LanguageBatchResult result = await AnalyticsClient.DetectLanguageAsync(new BatchInput(
-                    new List<Input>()
-                    {
-                        new Input("0", input)
-                    }));
-
-                if (result.Documents != null)
-                {
-                    languageResult.Language.Add("iso6391Name", result.Documents[0].DetectedLanguages[0].Iso6391Name);
-                    languageResult.Language.Add("name", result.Documents[0].DetectedLanguages[0].Name);
-                    languageResult.Language.Add("score", result.Documents[0].DetectedLanguages[0].Score.ToString());
-                }
+                DetectLanguageResultCollection result = await AnalyticsClient.DetectLanguageBatchAsync(
+                new List<string>() { input });
 
-                if (result.Errors != null)
+                if (!result[0].HasError)
                 {
-                    // Just return the empty Dictionary
-                }
+                    languageResult.Language.Add("iso6391Name", result[0].PrimaryLanguage.Iso6391Name);
+                    languageResult.Language.Add("name", result[0].PrimaryLanguage.Name);
+                    languageResult.Language.Add("score", result[0].PrimaryLanguage.ConfidenceScore.ToString());
+                }   
             }
 
             return languageResult;
```

Comment:
why the need of a Dictionary? Could you just return the `DetectedLanguage ` object?

Reply:
Origin version uses this pattern.

Reply:
https://github.com/Azure-Samples/cognitive-services-csharp-call-center/pull/2#discussion_r528565984